Readme.md

```diff
@@ -24,25 +61,68 @@ Fetch the [postgres](https://hub.docker.com/_/postgres) docker image before runn
 docker pull postgres
 ```
 
-## API
+### Worldcoin id contracts
+Worldcoin id contracts are ethereum smart contracts that are used by the sequencer  
 
+Clone [worldcoin id contracts](https://github.com/worldcoin/world-id-contracts) and execute `make build && npm install`  
+
+You will need a local ethereum chain, like geth or ganache.  
+Start up your local ethereum network and generate an account.  
+Go to your worldcoin id contracts repo folder and do `make deploy`. Follow the steps provided by the script.  
+Make sure you use a private key from your local ethereum network account.  
+!! Make a note of all the addresses generated by the script !!  
+
+### Semaphore-mtb
+Semaphore-mtb is a service for batch processing of Merkle tree updates.
+
+Clone [semaphore-mtb](https://github.com/worldcoin/semaphore-mtb) and execute `go build .` (you will need a golang compiler)  
+
+Go build will create a executable named gnark-mbu.  If you wen't trough the worldcoin id contracts deploy script,  
+you will have a generated keys file that is used by semaphore-mtb.
+```shell
+./gnark-mbu start --keys-file path/to/world-id-contracts/mtb/keys
+```
 
-## Database
+### Database
 
 ```shell
 docker run --rm -ti -p 5432:5432 -e POSTGRES_PASSWORD=password postgres
 ```
 
-## Hints
+Now you are ready to start up sequencer service!  
+```shell
+TREE_DEPTH=*your tree depth (eg. 16)* cargo run -- --batch-size *batch size for semaphore-mtb (eg. 3)* --batch-timeout-seconds 10 --database postgres://postgres:password@0.0.0.0:5432 --identity-manager-address *address from worldcoin id contracts identity manager* 
+--signing-key *private key you used to deploy smart contracts*
+```
 
-Lint, build, test, run
+## Tests
+
+Lint, build, test
 
 ```shell
-cargo fmt && cargo clippy --all-targets --features "bench, mimalloc" && cargo build --all-targets --features "bench, mimalloc" && cargo test --all-targets --features "bench, mimalloc" && cargo run --
+cargo fmt && cargo clippy --all-targets --features "bench, mimalloc" && cargo build --all-targets --features "bench, mimalloc" && cargo test --all-targets --features "bench, mimalloc"
 ```
 
+## Benchmarking
```

Comment:
The benchmarking stuff is largely out of date. We don't have any benchmarks in code, and in fact we don't have any usage of the `bench` feature in our code either.

We should probably remove it until we actually prepare a benchmarking suite of sorts.

Reply:
removed